### ISSUE TITLE
Median incorrectly calculated

### ISSUE THREAD

**Issue by semanticphilosopher** [DeltaQ-SD/deltaq#99]
The median (0.5 quantile) is incorrectly calculated in certain circumstances.

given 
```hs
--
-- ∆Q using Rationals
--
import DeltaQ

-- needed for visualisation within iHaskell
import Graphics.Rendering.Chart.Renderable

-- useful for rendering Rationals with 'just enough' precision
asFloat :: Rational -> Float
asFloat = fromRational

ex n = foldr1 sequentially (replicate n $ uniform 0 1) :: DQ

toRenderable $ plotCDFs "test" [("ex1", ex 1), ("ex2", ex 2), ("ex3", ex 3)]
toRenderable <$> plotCDFWithQuantiles "ex 3" [0.25, 0.5, 0.75] $ ex 3
asFloat <$> quantile (ex 1) 0.5
asFloat <$> quantile (ex 2) 0.5
asFloat <$> quantile (ex 3) 0.49999
asFloat <$> quantile (ex 3) 0.5
asFloat <$> quantile (ex 3) 0.50001
```
you get the following output:

![test](https://github.com/user-attachments/assets/4d19932f-caae-4ae1-b67a-a0efb2b414a3)
![ex 3](https://github.com/user-attachments/assets/5450bb41-a047-45aa-8dc3-903e67de053a)
```
Occurs 0.5
Occurs 1.0
Occurs 1.4999866
Occurs 2.0
Occurs 1.5000134
```

Could this be a manifestation of #69 ?

**Comment by HeinrichApfelmus** [comment]
> Could this be a manifestation of #69 ?

Yes, I think so. The symptoms are pretty much the ones reported: The root finding algorithm runs discards the middle of the interval (t = `3/2`) and runs off to an end of the interval instead.

As this particular example does not involve a double root, Peter's original fix should do the trick, though.

**Comment by HeinrichApfelmus** [comment]
I have merged pull request #105 to address your immediate problem #69 .

There are other circumstances in which `quantile` may give incorrect results due to a root of higher multiplicity (i.e. where the horizontal line is tangent to the graph of the cumulative probability distribution). I have recorded this issue in #104 .

**Comment by semanticphilosopher** [comment]
This resolves the immediate concern - thank you
